addons/web/static/src/views/fields/relational_utils.js

```diff
@@ -353,11 +355,35 @@ export class Many2XAutocomplete extends Component {
         });
     }
     mapRecordToOption(record) {
+        const label = record.__display_name_format && record.__display_name_format.split("\n")[0];
         return {
             value: record.id,
-            label: record.display_name ? record.display_name.split("\n")[0] : _t("Unnamed"),
+            label: label ? this.formatText(label) : _t("Unnamed"),
         };
     }
+
+    formatText(text) {
+        const boldEx = /\*\*(.+)\*\*/g;
+        const textMutedEx = /--(.+)--/g;
+        const tagEx = /&#x60;(.+)&#x60;/g;
+        const brEx = /&lt;br&gt;/g;
+        const tabEx = /&lt;tab&gt;/g;
+
+        text = htmlEscape(text);
+        return markup(
```

Comment:
ok to have it there for now, and we'll move it to some utils later if the feature is used somewhere else